constraint-solver/src/solver/exhaustive_search.rs

```diff
@@ -13,7 +14,9 @@ use std::hash::Hash;
 use super::Error;
 
 /// The maximum number of possible assignments to try when doing exhaustive search.
-const MAX_SEARCH_WIDTH: u64 = 1 << 12;
+const MAX_SEARCH_WIDTH: u64 = 1 << 10;
+/// The maximum range width of a variable to be considered for exhaustive search.
+const MAX_VAR_RANGE_WIDTH: u64 = 5;
```

Comment:
curious: what is the reasoning for this after lowering `MAX_SEARCH_WIDTH`?

Reply:
Basically, we'll want to make exhaustive search as restrictive as possible without causing any regression (which I assume would be caught by the tests). I haven't done any rigorous testing here.

But some intuition: I think we sometimes need to search over many boolean variables, and with the current `MAX_SEARCH_WIDTH`, we can include up to 10. But with a width of $2^{10}$, this could also include byte-constrained variables. But it's usually not helpful to search through all possible assignments of a byte. The new `MAX_VAR_RANGE_WIDTH` constant makes sure we never include variables with a large range.